File: EdgeWeightedDigraph.java
public class EdgeWeightedDigraph extends EdgeWeightedGraph {

  public EdgeWeightedDigraph() {
    super();
  }

  public EdgeWeightedDigraph(String filename) {
    super(filename);
  }

  @Override
  public void addEdge(String v, String w, double weight) {
    Edge e = new Edge(v, w, weight);
    addToList(v, e);
    if (!vertices.contains(v)) {
      vertices.add(v);
      totalVertices++;
    }
    if (!vertices.contains(w)) {
      vertices.add(w);
      totalVertices++;
    }
    totalEdges += 2;
    totalEdges++;
  }

  @Override
  public String toDot() {
    StringBuilder sb = new StringBuilder();
    sb.append("digraph {" + NEWLINE);
    sb.append("rankdir = LR;" + NEWLINE);
    sb.append("node [shape = circle];" + NEWLINE);
    for (Edge e : getEdges())
      sb.append(String.format("\"%s\" -> \"%s\" [label=\"%.3f\"]", e.getV(), e.getW(), e.getWeight()) + NEWLINE);
    sb.append("}" + NEWLINE);
    return sb.toString();
  }
}

File: edgeweighteddigraph.py
from edgeweightedgraph import EdgeWeightedGraph

class EdgeWeightedDigraph(EdgeWeightedGraph):

    def addEdge(self, v, w, weight):
        e = [v, w, weight]
        super().addToList(v, e)

    def toDot(self):
        NEWLINE = '\n'
        sb = "digraph {" + NEWLINE
        sb += "rankdir = LR;" + NEWLINE
        sb += "node [shape = circle];" + NEWLINE
        for v in sorted(self.getVerts()):
            for e in self.getAdj(v):
                sb += f'{e[0]} -- {e[1]} [label="{e[2]}"]' + NEWLINE
        sb += "}" + NEWLINE
        return sb

if __name__ == "__main__":

    #    g = EdgeWeightedDigraph()

    #    g.addEdge("0", "1")
    #    g.addEdge("0", "2")
    #    g.addEdge("2", "1")

    g = EdgeWeightedDigraph("exemplos/tinyEWD.txt")

    for v in g.getVerts():
        print(f"{v}: ", end="")
        for w in g.getAdj(v):
            print(f"{w} ", end="")
        print()
    print()
    print(g.toDot())
